app/components/articleSearch/records.tsx

```diff
@@ -1,18 +1,47 @@
-import { TypedRecord, makeTypedFactory } from "typed-immutable-record";
-import { IArticlesRecord } from "../../model/article";
+import { TypedRecord, makeTypedFactory, recordify } from "typed-immutable-record";
 import { List } from "immutable";
+import { IPapersRecord } from "../../model/paper";
 
 export enum SEARCH_SORTING {
   RELEVANCE,
   LATEST,
 }
 
+export interface ISearchItemInfo {
+  commentInput: string;
+  isAbstractOpen: Boolean;
+  isCommentsOpen: Boolean;
+}
+
+export interface ISearchItemInfoRecord extends TypedRecord<ISearchItemInfoRecord>, ISearchItemInfo {}
+
+const initialSearchItemInfo: ISearchItemInfoRecord = recordify({
+  commentInput: "",
+  isAbstractOpen: false,
+  isCommentsOpen: false,
+});
+
+export interface ISearchItemsInfo extends List<ISearchItemInfoRecord> {}
+
+export function initializeSearchItemsInfo(size: number): ISearchItemsInfo {
+  let searchItemInfoArray = [];
+
+  for (let i = 0; i < size; i++) {
```

Comment:
You can do it via map function.

Reply:
I think map is unnecessary because this function does not use any information of Papers.
This function only needs size of paper items